sources/packages/backend/apps/api/src/services/education-program-offering/_tests_/unit/education-program-offering.service-getCalculatedStudyBreaksAndWeeks.spec.ts

```diff
@@ -0,0 +1,247 @@
+          breakEndDate: "2023-06-13",
+          breakStartDate: "2023-06-08",
+          eligibleBreakDays: 6,
+          ineligibleBreakDays: 0,
+        },
+      ],
+      sumOfTotalEligibleBreakDays: 15,
+      sumOfTotalIneligibleBreakDays: 0,
+      totalDays: NaN,
+      totalFundedWeeks: NaN,
+      unfundedStudyPeriodDays: NaN,
+    });
+  });
+
+  it("Should not calculate study breaks when start study break date is not available.", () => {
+    //Arrange
+    const offeringStudyBreakCalculationContext: OfferingStudyBreakCalculationContext =
+      {
+        studyStartDate: "",
+        studyEndDate: "",
+        studyBreaks: [
+          {
+            breakStartDate: "",
+            breakEndDate: "2023-06-06",
+          },
+        ],
+      };
+
+    // Act
+    const calculatedStudyBreaksAndWeeks =
+      EducationProgramOfferingService.getCalculatedStudyBreaksAndWeeks(
+        offeringStudyBreakCalculationContext,
+      );
+
+    // Assert
+    expect(calculatedStudyBreaksAndWeeks).toEqual({
+      allowableStudyBreaksDaysAmount: NaN,
+      fundedStudyPeriodDays: NaN,
+      studyBreaks: [],
+      sumOfTotalEligibleBreakDays: 0,
+      sumOfTotalIneligibleBreakDays: 0,
+      totalDays: NaN,
+      totalFundedWeeks: NaN,
+      unfundedStudyPeriodDays: NaN,
+    });
+  });
+
+  it("Should not calculate study breaks when end study break date is not available.", () => {
+    //Arrange
+    const offeringStudyBreakCalculationContext: OfferingStudyBreakCalculationContext =
+      {
+        studyStartDate: "",
+        studyEndDate: "",
+        studyBreaks: [
+          {
+            breakStartDate: "2023-06-06",
+            breakEndDate: "",
+          },
+        ],
+      };
+
+    // Act
+    const calculatedStudyBreaksAndWeeks =
+      EducationProgramOfferingService.getCalculatedStudyBreaksAndWeeks(
+        offeringStudyBreakCalculationContext,
+      );
+
+    // Assert
+    expect(calculatedStudyBreaksAndWeeks).toEqual({
+      allowableStudyBreaksDaysAmount: NaN,
+      fundedStudyPeriodDays: NaN,
+      studyBreaks: [],
+      sumOfTotalEligibleBreakDays: 0,
+      sumOfTotalIneligibleBreakDays: 0,
+      totalDays: NaN,
+      totalFundedWeeks: NaN,
+      unfundedStudyPeriodDays: NaN,
+    });
+  });
+
+  it("Should calculate eligible break days as a maximum of 21 days when a study break exceeds it.", () => {
+    //Arrange
+    const offeringStudyBreakCalculationContext: OfferingStudyBreakCalculationContext =
+      {
+        studyStartDate: "2023-05-29",
+        studyEndDate: "2023-09-29",
+        studyBreaks: [
+          {
+            breakStartDate: "2023-06-08",
+            breakEndDate: "2023-06-30",
+          },
+        ],
+      };
+
+    // Act
+    const calculatedStudyBreaksAndWeeks =
+      EducationProgramOfferingService.getCalculatedStudyBreaksAndWeeks(
+        offeringStudyBreakCalculationContext,
+      );
+
+    // Assert
+    expect(calculatedStudyBreaksAndWeeks).toEqual({
```

Comment:
Is there a benefit of using `toStrictEqual` instead `toEqual` for these cases?

Reply:
In this case we can use `.toStrictEqual` because the service is creating an object literal. 

Benefits of using  `.toEqual`:

- keys with undefined properties are checked, e.g. {a: undefined, b: 2} will not equal {b: 2};
- undefined items are taken into account, e.g. [2] will not equal [2, undefined];
- array sparseness is checked, e.g. [, 1] will not equal [undefined, 1];
- object types are checked, e.g. a class instance with fields a and b will not equal a literal object with fields a and b.

I changed the comparions to `.toStrictEqual`. Thanks!